src/main.js

```diff
@@ -234,29 +210,39 @@ function SearchResultCollection(includedNodeLists, excludedNodeLists) {
     }
   });
   this.focusedIndex = 0;
-  this.focus = function(index, scrollToResult = true) {
+  this.getHighlightedElement = function (index) {
+    if (extension.searchEngine.HighlightedParentSelector !== undefined) {
+        return this.items[index].anchor.closest(extension.searchEngine.HighlightedParentSelector)
+    } else {
+        return this.items[index].anchor
+    }
+  }
+  this.focus = function (index, scrollToResult = true) {
     if (this.focusedIndex >= 0) {
-      let item = this.items[this.focusedIndex];
+      let item = this.getHighlightedElement(this.focusedIndex);
       // Remove focus outline from previous item.
-      item && item.anchor.classList.remove('focused-search-result');
-      item && item.anchor.classList.remove('no-outline');
+      item.classList.remove(extension.searchEngine.HighlightClass);
+      item.classList.remove('no-outline');
     }
-    const newItem = this.items[index];
+    const Highlighted = this.getHighlightedElement(index);
```

Comment:
Should be lowercase: `highlighted`.
Please try to follow the Google Javascript style guide: https://google.github.io/styleguide/jsguide.html

(And I should probably mention it in the README...)

Reply:
Alright, just updated my eslintrc